core/include/mutex.h

```diff
@@ -110,15 +110,12 @@
 
 #include <stddef.h>
 #include <stdint.h>
+#include <stdbool.h>
 
 #include "kernel_defines.h"
 #include "list.h"
 #include "thread.h"
 
-#ifndef __cplusplus
-#include "irq.h"
```

Comment:
CI found some users that relied on this implicit include. Better add the include to the .c file that makes use of it

Reply:
Looks like there are still some missing.

Reply:
`bme680_hal.c` also needs an update